File: matlab/autd3/Null.m
%{
%File: Null.m
%Project: autd3
%Created Date: 10/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 10/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef Null < Gain

    methods

        function obj = Null()
            obj = obj@Gain();
            pp = libpointer('voidPtrPtr', obj.ptr);
            calllib('autd3capi', 'AUTDGainNull', pp);
        end

    end

end

File: julia/src/Null.jl
# File: Null.jl
# Project: src
# Created Date: 14/06/2022
# Author: Shun Suzuki
# -----
# Last Modified: 14/06/2022
# Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
# -----
# Copyright (c) 2022 Shun Suzuki. All rights reserved.
# 

struct Null
    _gain::Gain
    _body_ptr
    function Null()
        chandle = Ref(Ptr{Cvoid}(0))
        autd3capi.autd_gain_null(chandle)
        g = Gain(chandle[])
        new(g, chandle[])
    end
end
